projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/trait-wpcom-rest-api-proxy-request-trait.php

```diff
@@ -23,7 +23,7 @@ trait WPCOM_REST_API_Proxy_Request_Trait {
 	public function proxy_request_to_wpcom_as_user( $request, $path = '' ) {
 		$blog_id = \Jetpack_Options::get_option( 'id' );
 		$path    = '/sites/' . rawurldecode( $blog_id ) . rawurldecode( $this->rest_base ) . ( $path ? '/' . rawurldecode( $path ) : '' );
-		$api_url = add_query_arg( $request->get_query_params(), $path );
+		$api_url = remove_query_arg( 'rest_route', add_query_arg( $request->get_query_params(), $path ) );
```

Comment:
I was wondering if we could be a bit more verbose about the change, so we remember why we did that a few months from now?

Something like this?

```suggestion
		$query_params = $request->get_query_params();

		/*
		 * A rest_route parameter can be added when using plain permalinks.
		 * It is not necessary to pass them to WordPress.com,
		 * and may even cause issues with some endpoints.
		 * Let's remove it.
		 */
		if ( isset( $query_params['rest_route'] ) ) {
			unset( $query_params['rest_route'] );
		}
		$api_url = add_query_arg( $query_params, $path );
```

What do you think?

Reply:
Agreed a 100% 🤝